pkg/skaffold/kubernetes/watcher.go

```diff
@@ -17,15 +17,49 @@ limitations under the License.
 package kubernetes
 
 import (
+	"errors"
 	"fmt"
 
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/watch"
 )
 
-// AggregatePodWatcher returns a watcher for multiple namespaces.
-func AggregatePodWatcher(namespaces []string, aggregate chan<- watch.Event) (func(), error) {
-	watchers := make([]watch.Interface, 0, len(namespaces))
+type PodWatcher interface {
+	Register(receiver chan<- PodEvent)
```

Comment:
This should return a cleanup / unregister function?

Reply:
not sure it needs to return one necessarily since it's just tracking the receivers in a slice, but it doesn't look like those channels ever get cleaned up?

Reply:
Right now, there's no need for cleaning up. Might need it in step 2, though